File: 0137-single-number-ii/0137-single-number-ii.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function(nums) {
    const map = new Map();
    for(let i = 0; i<nums.length; i++){
        map.set(nums[i], 1 + (map.has(nums[i]) ? map.get(nums[i]) : 0))
    }
    for(let i of map.keys()){
        if(map.get(i) == 1) return i;
    }
    return -1;
};

File: 0137-single-number-ii/0137-single-number-ii.cpp
class Solution {
public:
    // time O(n) spaceO(n) solution
    int singleNumber(vector<int>& nums) {
        unordered_map<int,int> mp; // value - freq
        for(int& i : nums){
            if(mp.find(i) != mp.end()) mp[i]++;
            else mp.insert({i,1});
        }
        for(auto i : mp){
            if(i.second == 1) return i.first;
        }
        return -1;
    }
};